xapian-core/include/xapian/cluster.h

```diff
@@ -0,0 +1,633 @@
+ *  frequency
+ */
+class XAPIAN_VISIBILITY_DEFAULT TermListGroup : public FreqSource {
+
+  private:
+
+    /** Map of the terms and its corresponding term frequencies.
+     *  The term frequency of a term stands for the number of documents it indexes
+     */
+    std::unordered_map<std::string, doccount> termfreq;
+
+    /// Number of documents added to the termlist
+    doccount docs_num;
+
+    /** Add a single document and calculates its corresponding term frequencies
+     *
+     *  @param doc	Adds a document and updates the TermListGroup based on the
+     *			terms found in the document
+     */
+    void add_document(const Document &doc);
+
+  public:
+    /** Constructor
+     *
+     *  @params docs	MSet object used to construct the TermListGroup
+     */
+    explicit TermListGroup(const MSet &docs);
+
+    /** Return the number of documents that the term 'tname' exists in
+     *  or the number of documents that a certain term indexes
+     *
+     *  @param tname	The term for which we return the frequency value
+     */
+    doccount get_termfreq(const std::string &tname) const;
+
+    doccount get_doccount() const;
+};
+
+/** Abstract class representing a point in the VSM
+ */
+class XAPIAN_VISIBILITY_DEFAULT PointType
+    : public Xapian::Internal::opt_intrusive_base {
+
+  protected:
+
+    /** Implement a map to store the terms within a document
+     *  and their pre-computed TF-IDF values
+     */
+    std::unordered_map<std::string, double> values;
+
+    /// Store the squared magnitude of the PointType
+    double magnitude;
+
+    /** Set the value 'value' to the mapping of a term
+     *
+     *  @param term	Term for which the value is supposed
+     *			to be changed
+     *  @param value	The value to which the mapping of the
+     *			term is to be set
+     */
+    void set_value(const std::string &term, double value);
+
+  public:
+    /// Constructor
+    PointType() { magnitude = 0; }
+
+    /// Return a TermIterator to the beginning of the termlist
+    TermIterator termlist_begin() const;
+
+    /// Return a TermIterator to the end of the termlist
+    TermIterator termlist_end() const {
+	return TermIterator(NULL);
+    }
+
+    /** Validate whether a certain term exists in the termlist
+     *  or not by performing a lookup operation in the existing values
+     *
+     *  @param term	Term which is to be searched
+     */
+    bool contains(const std::string &term) const;
+
+    /** Return the TF-IDF weight associated with a certain term
+     *
+     *  @param term	Term for which TF-IDF weight is returned
+     */
+    double get_value(const std::string &term) const;
+
+    /** Add the value 'value' to the mapping of a term
+     *
+     *  @param term	Term to which the value is to be added
+     *  @param value	Value which has to be added to the existing
+     *			mapping of the term
+     */
+    void add_value(const std::string &term, double value);
```

Comment:
@ojwb Thanks for pointing this out.
The add_value method is used to add the TF-IDF weight of the points in a cluster. So these values could more specifically be called weights.
I guess changing the methods like `add_value, set_value and get_value` to `add_weight, set_weight and get_weight` would be better because its easier to read and conveys the right meaning.